pkg/virt-controller/watch/migration/migration_test.go

```diff
@@ -1419,6 +1421,46 @@ var _ = Describe("Migration watcher", func() {
 			expectMigrationCompletedState(migration.Namespace, migration.Name)
 		})
 
+		It("should not override the MigrationState of a completed migration when a new one is created", func() {
+			vmi := newVirtualMachine("testvmi", virtv1.Running)
+			addNodeNameToVMI(vmi, "node02")
+
+			completedMigration := newMigration("completed-migration", vmi.Name, virtv1.MigrationSucceeded)
+			runningMigration := newMigration("running-migration", vmi.Name, virtv1.MigrationRunning)
+
+			vmi.Status.MigrationState = &virtv1.VirtualMachineInstanceMigrationState{
+				MigrationUID:                   runningMigration.UID,
+				TargetNode:                     "node01",
+				SourceNode:                     "node02",
+				TargetNodeAddress:              "10.10.10.10:1234",
+				StartTimestamp:                 pointer.P(metav1.Now()),
+				EndTimestamp:                   pointer.P(metav1.Now()),
+				TargetNodeDomainReadyTimestamp: pointer.P(metav1.Now()),
+				Failed:                         false,
+				Completed:                      true,
+			}
+
+			addMigration(completedMigration)
+			addVirtualMachineInstance(vmi)
+			addPod(newSourcePodForVirtualMachine(vmi))
+			addPod(newTargetPodForVirtualMachine(vmi, completedMigration, k8sv1.PodRunning))
+
+			addMigration(runningMigration)
+
+			virtClientset.Fake.PrependReactor("update", "virtualmachineinstancemigrations/status", func(action testing.Action) (bool, runtime.Object, error) {
+				updateAction := action.(testing.UpdateAction)
+				updatedMigration := updateAction.GetObject().(*virtv1.VirtualMachineInstanceMigration)
+
+				if updatedMigration.Name == "completed-migration" {
+					Fail("UpdateStatus should not be called for the completed migration")
+				}
+
+				return false, nil, nil
+			})
```

Comment:
Instead of prepend reactor, how about querying the virtclientset after the controller execution?
We can better define the completed migration:
```go
	completedMigration := newMigration("completed-migration", vmi.Name, virtv1.MigrationSucceeded)
	completedMigration.Status.MigrationState = &virtv1.VirtualMachineInstanceMigrationState{
		MigrationUID: completedMigration.UID,
		TargetNode:   "node02",
		SourceNode:   "node01",
		Failed:       false,
		Completed:    true,
	}
	[...]
```
and then perform the check of the migration state after the controller execution:
```go
    [...]
	sanityExecute()

	oldMigration, err := virtClientset.KubevirtV1().VirtualMachineInstanceMigrations(completedMigration.Namespace).Get(context.TODO(), completedMigration.Name, metav1.GetOptions{})
	Expect(err).ToNot(HaveOccurred())
	Expect(oldMigration.Status.MigrationState.MigrationUID).To(Equal(completedMigration.UID))
})
```
WDYT?
Thanks

Reply:
Done